docs/configuration.md

```diff
@@ -0,0 +1,142 @@
+## TLA<sup>+</sup> formatter configuration
+
+The TLA<sup>+</sup> formatter is very flexible: the default used is described in this reference style guide, but each part is very configurable.
+It is possible to define a user level config, and a project level config. If there is no project-level config, the user-level one will be used instead.
+The config is serialized as a [toml](https://toml.io/en/) file.
+
+## Full example:
+This is the complete config file that can be copy-pasted and personalized. It's configured with the default values already.
+You can also use `tlaplusformatter --sample-config > ~/.tlaplus-formatter/config.toml` to kickstart your user level config.
+```
+[general]
+indentSizeSpaces = 4
+startDefinitonOnNewLine = true
+alignDefinitionWithDeclaration = true
+[keywords.module]
+Dashes = Ignore
+Equals = Ignore
+[keyword.extends]
+MultipleSeparateLines = true
+```
+
+
+```
+Template:
+## name / ``
+### Configuration
+This is the default setting:
+```
+```
+```
+
+## General settings / `general`
+These settings define general and default behaviors. Some of them can be tuned per keyword.
+```
+[general]
+indentSizeSpaces = 4
+startDefinitonOnNewLine = true
+alignDefinitionWithDeclaration = true
+```
+### `startDefinitonOnNewLine: boolean`
+The general default to start all definitions on a new line.
+* Possible value: `true` or `false`.
+* Default value: `true`.
+  Example with value `true`:
+```
+EXTENDS
+        TLC
+MyOperator == 
+              TRUE
+```
+Example with value `false`:
+```
+EXTENDS TLC
+MyOperator == TRUE
+```
+
+
+
+### `alignDefinitonWithDeclaration: boolean`
+Align or not the definitions with the declaration. When false, the next line will be indented by `indentSizeSpaces`.
+* Possible value: `true` or `false`.
+* Default value: `true`.
+  Example with value `true`:
+```
+MyOperator ==
+              TRUE
+HCnxt ==
+         hr' = IF
+                    hr # 12
+               THEN
+                    hr + 1
+               ELSE
+                    1
+```
+Example with value `false`, assuming `indentSizeSpaces` of 2:
+```
+MyOperator ==
+    TRUE
+HCnxt ==
+  hr' = IF
+          hr # 12
+        THEN
+          hr + 1
+        ELSE
+          1
+```
+
+## Module / `keywords.module`
+A module is defined as:
+* five or more dashes,
+* the keyword MODULE,
+* the module's name
+* Four or more dashes.
+  Four or more equal (=) symbols are used to define the end of the module.
+
+### Recommendation
+Use only 4 dashes and 4 equals:
+```
+---- MODULE ----
+====
+```
+
+### Configuration
+This is the default setting:
+```
+[keywords.module]
+Dashes = Ignore
+Equals = Ignore
```

Comment:
I often like to extend the number of characters in my `---- MODULE Name ----` line and `====` line to an equal number (78 seems to be standard), with the `MODULE Name` text centered; it would be nice to have this as an option.